### DISCUSSION TITLE
[Question] Can I extend other plugins using my own plugin?

### DISCUSSION THREAD

**Discussion by periabyte** [strapi/strapi#6066] [Legacy Topics]
**Describe the topic**
I'm currently creating a strapi plugin for our company, as we intend to use strapi as our main Headless CMS. I would like to implement our business logic using the plugin but the logic needs to extend (like what we'll do in `extensions` folder) the user-permissions plugin, is it possible?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by lauriejim** [comment]
Can you please precise what you are trying to do exactly, I'm not sure to understand.

**Reply by derrickmehaffy** [reply]
I believe they are intending to use their own plugin to modify a Strapi one (without hard coded extensions).

**Comment by lauriejim** [comment]
This is not possible for now. The only way to customize a plugin with another plugin is for the Content Manager to add/replace a field.
Here is the guide - [guide](https://strapi.io/documentation/3.0.0-beta.x/guides/registering-a-field-in-admin.html)

We will introduce a complete API in the future to let you have a programatic impact on plugins.
Without ejecting files.
